_posts/en/2017-01-25-ios-how-to-combine-delegates-and-closures.md

```diff
@@ -0,0 +1,250 @@
+---
+layout: post
+title: "[iOS] How to combine Delegates and Closures ?"
+authors:
+    - thuchon
+lang: en
+permalink: /ios-how-to-combine-delegates-and-closures/
+excerpt: "Let's make Delegates and Closures work together"
+date: '2017-01-25 14:30:42 +0100'
+date_gmt: '2017-01-25 13:30:42 +0100'
+categories:
+    - Mobile
+tags:
+    - ios
+    - delegates
+    - closure
+---
+
+### Introduction
+
+Hi Astronauts, today we will keep talking about mobile development, and as always native style.
+
+This article is following up with the 2 previous ones, and it's mandatory that you read them before in order to understand what's going on in this one:
+
+[Delegates VS Closures](https://blog.eleven-labs.com/en/delegates-closures/){:rel="nofollow noreferrer"}
+
+[Let's talk about listeners](https://blog.eleven-labs.com/en/android-listeners/){:rel="nofollow noreferrer"}
+
+
+If you have already read the 2 previous articles, I guess you already have an idea about what this one is about.
+- Yes we already know, so please hurry-up, we want to know how we can make something as sexy as the listeners but on iOS this time.
+- Ok, just a bit of technical chit-chat before, and then were are ready to go.
+
+**How are we going to proceed?:**
+
+As in the first article, so that everyone is happy I will provide some DUMMY code for both Objective-C and Swift.
+
+As you may know in the iOS universe, we can use both Delegates and Closures. Usually closure is used for more flexibility and it is also easier to implement. However, in some cases, graphic components for example are just made to work with a delegate or a datasource. Right now I'm thinking about 2 components: UITableView and UICollectionView.
+
+With those 2 components, you cannot use block/closure, and you have to use a good old delegate. Usually it's not really a big issue and you can just work like that. But let say that you have to work with many of those components on a single screen, in this case the code can start to be really. You'll find yourself with some huge classes, and it really starts to be difficult to produce some beautiful and elegant code. What I would like to show you today is a solution that is my mind is pretty clean, and easy to put in place
```

Comment:
the code can start to be really what? :) missing word

Reply:
let's say

Reply:
What I would like to show you today is a solution, that, in my opinion, is pretty clean, and easy to put in place.